### PR TITLE
Traefik v2

### PR DESCRIPTION
**Pull request by hbjydev**

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
I think it would be great if we would offer both Traefik and Traefik 2 as many people probably still didn't migrate.

File: manuscript/ha-docker-swarm/traefik.md [hunk 1]
```diff
@@ -52,42 +52,40 @@ While it's possible to configure traefik via docker command arguments, I prefer
...
+[global]
+  checkNewVersion = true
+
+# Enable the Dashboard
+[api]
+
+# Write out Traefik logs
+[log]
+  level = "DEBUG"
+  filePath = "/traefik.log"
```
Comment (SuperSandro2000) [inline manuscript/ha-docker-swarm/traefik.md:64]: This is probably not what we want.

Comment (itshaydendev) [reply manuscript/ha-docker-swarm/traefik.md:64]: No, that's the translation of the `--log.level=DEBUG` command-line option that was passed.

File: manuscript/ha-docker-swarm/traefik.md [hunk 2]
```diff
@@ -133,8 +131,7 @@ version: "3"
 
 services:
   traefik:
-    image: traefik
-    command: --web --docker --docker.swarmmode --docker.watch --docker.domain=example.com --logLevel=DEBUG
+    image: traefik:v2.2
```
Comment (SuperSandro2000) [inline manuscript/ha-docker-swarm/traefik.md:134]:
```suggestion
    image: traefik:v2
```

**Comment by itshaydendev** [comment]
Well, perhaps for that the solution should just be going back a tag/going back in the branch.
With outdated guides comes outdated software, and I'm sure I don't need to explain that it's a bad move to promote outdated software.

File: manuscript/ha-docker-swarm/traefik.md (hunk 2 above)
Comment (itshaydendev) [reply manuscript/ha-docker-swarm/traefik.md:134]: Always better to stick to specific versions. Syntax could change.

File: manuscript/recipes/mail.md [hunk 3]
```diff
@@ -130,6 +130,8 @@ services:
       - internal
     deploy:
       replicas: 1
+      labels:
+        - traefik.enable=false
```
Comment (SuperSandro2000) [inline manuscript/recipes/mail.md:134]: This is only need if auto discovery is on.

File: manuscript/ha-docker-swarm/traefik.md (hunk 2 above)
Comment (itshaydendev) [reply manuscript/ha-docker-swarm/traefik.md:134]: Actually, thinking about *this* one, it should be okay.

File: manuscript/recipes/mail.md (hunk 3 above)
Comment (itshaydendev) [reply manuscript/recipes/mail.md:134]: Which it is by default, and was in the old configuration.

**Comment by itshaydendev** [comment]
@SuperSandro2000 What?

**Comment by SuperSandro2000** [comment]
> Well, perhaps for that the solution should just be going back a tag/going back in the branch.

You can't to this on the website.

> With outdated guides comes outdated software

I wouldn't consider Traefik 1 outdated. Traefik 1 is far easier to use than Traefik 2 which falls short in many cases and I don't think we are at a point where we should purge it from every doc.

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000 What?

If you add review comments you are enforced to also add a comment which is annoying.

**Comment by funkypenguin** [comment]
Thanks for all the input, guys! 👍  I'd be happy to see two recipes (_Traefik and Traefik2_) come out of this, but I'd agree that Traefik v1.7 should stick around, since it's a simpler installation, and quite adequate for some users.

**Comment by funkypenguin** [comment]
Touching base here - this PR is still flagged as a WIP - is the W still IP ? :)

**Comment by itshaydendev** [comment]
Sorry, work has ramped up irl, not been able to do much work on this. Especially given the want to keep v1 and v2 configs.

**Comment by hbjydev** [comment]
Closing this, it's gotten way too old for me to really finish working on, even now I have a bit of time on my hands.

**Pull request closed without merging**